ietf-layer0-types-ext.yang

```diff
@@ -923,4 +966,23 @@ module ietf-layer0-types-ext {
       description "Wavelength Allocation Method";
     }
   }
+
+  grouping frequency-range {
+    description
+      "The parameters that define a frequency range.";

```

Comment:
should there be validation logic to make sure upper-frequency >= lower-frequency, given that both parameters are set mandatory=true?

Reply:
The grouping so far has been used to represent cases where upper-frequency > lower-frequency

Reply:
I mean that to add a "must" statement in the leaf of upper-frequency to validate the case above

Reply:
I agree. I was just wondering whether the must statement should validate that upper-frequency > lower-frequency rather than upper-frequency >= lower-frequency

Reply:
ok I see. In the case that both upper- and lower-frequency are mandated it is better to use > rather than >=. I was thinking of cases where the upper-frequency could be optional, which represents a frequency range from a lower-frequency to whatever is supported at the upper end, but I am not sure whether it is a case we need to consider. I checked openconfig and in there where lower- and upper-frequency are specified, both of of the two parameters are optional and not mandatory. perhaps @dieterbeller may also comment -

Reply:
> ok I see. In the case that both upper- and lower-frequency are mandated it is better to use > rather than >=.
> 
> Also I was thinking of cases where the upper-frequency could be optional, which represents a frequency range from a lower-frequency to whatever is supported at the upper end, but I am not sure whether it is a case we need to consider. I checked openconfig and in there where lower- and upper-frequency are specified, both of of the two parameters are optional and not mandatory. perhaps @dieterbeller may also comment -

I do not remind who , but someone in our group commented that the two parameters should be "mandatory", this is why we put mandatory to the 2 leaves.

Reply:
The frequency range is used in the amp to define per band attributes: this is mandatory to know, otherwise, it is not possible to identify which amp is for which band.

Reply:
Addressed in https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext/pull/25/commits/131b8f8939b6aa5cd6ed79608ff5263589ef687a